### DISCUSSION TITLE
Support for filtering by camera and lens

### DISCUSSION THREAD

**Discussion by djjudas21** [photoprism/photoprism#2098] [Ideas]
I shoot on many different cameras and lenses, and it would be a valuable feature for me to be able to filter my photos by camera and lens. I'm thinking something in the same style as the Labels view.

**Comment by graciousgrey** [comment]
You mean one generated album for each camera+lens combination?

**Reply by djjudas21** [reply]
Not quite. I think Camera and Lens should be separate categories.

Under Camera I could filter by "iPhone X" or "Canon EOS 80D" and under Lens I could filter by "iPhone X telephoto lens", "Canon EF 50mm" or "Canon EF 18-135mm".

These values are all available in EXIF data.

The Lens view would probably have more benefit for photographers who are using interchangeable-lens cameras, although most phones these days do also have multiple lenses on them.

**Reply by graciousgrey** [reply]
PhotoPrism reads this metadata and you can [filter for camera](https://docs.photoprism.app/user-guide/organize/search/#search-filter) already using the ui (e.g.: https://demo.photoprism.app/browse?view=cards&camera=6&order=newest&public=true&quality=3). A search filter for the lens name is on our [to do](https://github.com/photoprism/photoprism/issues/2079).

**Reply by djjudas21** [reply]
Oh that's cool, thanks, I'll check this out.

I think it would still be good to have cameras and lenses as browsable objects though.

**Reply by 2600box** [reply]
Very much agree. This has always been a core way that I browse and organise my libraries by which camera took the photo. This can allow me to identify the photographer for example.

**Reply by graciousgrey** [reply]
camera and lens filters are available in our preview already: https://docs.photoprism.app/user-guide/organize/search/#search-filter-overview
